src/main/java/com/zufar/onlinestore/user/api/UserService.java

```diff
@@ -0,0 +1,39 @@
+package com.zufar.onlinestore.user.api;
+
+import com.zufar.onlinestore.user.converter.UserDtoConverter;
+import com.zufar.onlinestore.user.dto.UserDto;
+import com.zufar.onlinestore.user.entity.UserEntity;
+import com.zufar.onlinestore.user.exception.UserNotFoundException;
+import com.zufar.onlinestore.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UserService implements UserApi {
+
+    private final UserRepository userCrudRepository;
+    private final UserDtoConverter userDtoConverter;
+
+    @Override
+    public UserDto saveUser(final UserDto saveUserRequest) {
+        UserEntity userEntity = userDtoConverter.toEntity(saveUserRequest);
+        UserEntity userEntityWithId = userCrudRepository.save(userEntity);
+        return userDtoConverter.toDto(userEntityWithId);
+    }
+
+    @Override
+    public UserDto getUserById(final UUID userId) throws UserNotFoundException {
```

Comment:
@ Override
public UserDto getUserById(final UUID userId) throws UserNotFoundException {
    return userCrudRepository.findById(userId)
        .map(userDtoConverter::toDto)
        .orElseThrow(() -> {
            log.warn("Failed to get the user with the userId = {}.", userId);
            return new UserNotFoundException(userId);
        });
}